src/trackedit/internal/au3/au3interaction.cpp

```diff
@@ -117,6 +117,194 @@ muse::Ret Au3Interaction::canPasteClips(const std::vector<TrackId>& dstTracksIds
     return muse::make_ok();
 }
 
+muse::Ret Au3Interaction::makeRoomForClip(const ClipKey &clipKey)
+{
+    trackedit::ITrackeditProjectPtr prj = globalContext()->currentTrackeditProject();
+
+    WaveTrack* waveTrack = DomAccessor::findWaveTrack(projectRef(), ::TrackId(clipKey.trackId));
+    IF_ASSERT_FAILED(waveTrack) {
+        return false;
+    }
+
+    std::shared_ptr<WaveClip> clip = DomAccessor::findWaveClip(waveTrack, clipKey.clipId);
+    IF_ASSERT_FAILED(clip) {
+        return false;
+    }
+
+    std::list<std::shared_ptr<WaveClip> > clips = DomAccessor::waveClipsAsList(waveTrack);
+
+    clips.sort([](const std::shared_ptr<WaveClip>& c1, const std::shared_ptr<WaveClip>& c2) {
+        return c1->GetPlayStartTime() < c2->GetPlayStartTime();
+    });
+
+    for (const auto& otherClip : clips) {
+        if (clip == otherClip) {
+            //! NOTE do not modify the clip the action was taken on
+            continue;
+        }
+
+        //! NOTE check if clip hovers whole other clip
+        if (clip->GetPlayStartTime() <= otherClip->GetPlayStartTime()
+            && clip->GetPlayEndTime() >= otherClip->GetPlayEndTime()) {
```

Comment:
unsafe comparison of fractional values 
Options: 
* use functions from muse realfn (muse::RealIsEqualOrLess ...)
* added local variable with type secs_t 
* do like this `if (secs_t(clip->GetPlayStartTime()) <= secs_t(otherClip->GetPlayStartTime())) `
* change type for clip->GetPlayStartTime()  and clip->GetPlayEndTime() to secs_t